src/main/kotlin/stringcalculator/entity/Formula.kt

```diff
@@ -0,0 +1,37 @@
+package stringcalculator.entity
+
+class Formula private constructor(
+    val delimiter: Regex = DEFAULT_DELIMITER_REGEX,
+    val expression: String
+) {
+    companion object {
+        private val CUSTOM_DELIMITER_REGEX = "//(.)\n(.*)".toRegex()
+        private val DEFAULT_DELIMITER_REGEX = "[,:]".toRegex()
+
+        fun of(input: String?): Formula {
+            if (input.isNullOrEmpty()) {
+                return Formula(expression = "0")
+            }
+            return CUSTOM_DELIMITER_REGEX.find(input)?.let {
+                buildCustomDelimiterFormula(it)
+            } ?: buildDefaultDelimiterFormula(input)
+        }
+
+        private fun buildCustomDelimiterFormula(matchResult: MatchResult): Formula {
+            require(matchResult.groupValues.size == 3)
+            val customDelimiter = matchResult.groupValues[1].toRegex()
+            val expression = matchResult.groupValues[2]
```

Comment:
customDelimiter가 String 타입이라 별도의 Regex 타입으로 변환하는 작업이 필요하겠지만, 이런식으로 [구조분해(Destructuring declarations)](https://kotlinlang.org/docs/destructuring-declarations.html)를 활용해볼 수도 있을 것 같아요! 😃 

```suggestion
            val (_, customDelimiter, expression) = matchResult.groupValues
```

구조분해를 사용하는 방식으로 바꾸는게 더 낫다는 의미는 아니고, 코틀린에서는 구조분해 기능을 제공한다는 정보를 전달드리고자 가볍게 얘기봤어요. 😉

Reply:
@Rok93 훨씬 좋아보입니다.